parsers/regex/regex.go

```diff
@@ -0,0 +1,165 @@
+// Package regex consumes logs given user-defined regex format for lines
+
+/* Example usage (from honeytail root)
+  		go install . && \
+ 			honeytail -p regex -k $HONEYTAIL_WRITEKEY \
+			-f /private/var/system.log \
+			--dataset 'MY_TEST_DATASET' \
+			--backfill \
+			--regex.line_regex="(?P<time>\d{2}:\d{2}:\d{2})" \
+			--regex.timefield="time" \
+			--regex.time_format="%H:%M:%S"
+*/
+
+package regex
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/httime"
+	"github.com/honeycombio/honeytail/parsers"
+)
+
+type Options struct {
+	LineRegex       string `long:"line_regex" description:"a regular expression with named capture groups representing the fields you want parsed"`
+	TimeFieldName   string `long:"timefield" description:"Name of the field that contains a timestamp"`
+	TimeFieldFormat string `long:"time_format" description:"Timestamp format to use (strftime and Golang time.Parse supported)"`
+
+	NumParsers int `hidden:"true" description:"number of regex parsers to spin up"`
+}
+
+type Parser struct {
+	conf       Options
+	lineParser parsers.LineParser
+}
+
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
+
+	// Regex can't be blank
+	if p.conf.LineRegex == "" {
+		logrus.Debug("LineRegex is blank")
+		return errors.New("LineRegex is blank")
+	}
+
+	// Compile regex
+	if p.conf.LineRegex != "" {
+		lineRegex, err := regexp.Compile(p.conf.LineRegex)
+		if err != nil {
+			logrus.WithFields(logrus.Fields{
+				"LineRegex": p.conf.LineRegex,
+			}).Debug("Could not compile LineRegex")
+			return err
+		}
+
+		// Require at least one named group
+		var numNamedGroups int
+		for _, groupName := range lineRegex.SubexpNames() {
+			if groupName != "" {
+				numNamedGroups++
+			}
+		}
+		if numNamedGroups == 0 {
+			logrus.WithFields(logrus.Fields{
+				"LineRegex": p.conf.LineRegex,
+			}).Error("No named capture groups")
+			return errors.New("No named capture groups found; must provide at least one named group")
```

Comment:
Here again, I wonder if we can make this clearer for users. Right now this'll print 
```
INFO[0000] Starting honeytail                           
ERRO[0000] No named capture groups                       LineRegex=.*
FATA[0000] err initializing parser module                err=No named capture groups found; must provide at least one named group parser=regex
```

which is a bit confusing to me. Could we return an error value `No named capture groups found in the regex <regex>. Your regex must contain at least one named group`? Or something like that?